documents/CMakeLists.txt

```diff
@@ -22,13 +23,62 @@ if(DOXYGEN_FOUND)
     configure_file(${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile.in ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile)
     add_custom_target(MaterialXDocs ${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile
                       WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-                      COMMENT "Generating HTML documentation: ${DOXYGEN_HTML_OUTPUT_DIR}/index.html")
+                      COMMENT "Building MaterialX C++ API Documentation: ${DOXYGEN_HTML_OUTPUT_DIR}/index.html")
     add_custom_command(TARGET MaterialXDocs PRE_BUILD
                        COMMAND ${CMAKE_COMMAND} -E copy_directory
-                       ${CMAKE_SOURCE_DIR}/documents/DoxygenAwesome ${CMAKE_CURRENT_BINARY_DIR})
+                       ${CMAKE_SOURCE_DIR}/documents/DoxygenAwesome ${DOXYGEN_HTML_OUTPUT_DIR})
     add_custom_command(TARGET MaterialXDocs PRE_BUILD
                        COMMAND ${CMAKE_COMMAND} -E copy_directory
-                       ${CMAKE_SOURCE_DIR}/documents/Images ${CMAKE_CURRENT_BINARY_DIR})
+                       ${CMAKE_SOURCE_DIR}/documents/Images ${DOXYGEN_HTML_OUTPUT_DIR})
     install(DIRECTORY ${DOXYGEN_HTML_OUTPUT_DIR}
             DESTINATION "documents" MESSAGE_NEVER)
+else()
+    message(WARNING "Unable to build MaterialX C++ API Documentation: Doxygen not found.")
 endif(DOXYGEN_FOUND)
+
+
+# MaterialX Python API Documentation
```

Comment:
Wanted to double check if there is any dependence on the actual building the Python modules as that is a build option?

I assume not, but just checking.

Reply:
Yes indeed, good point -- there is!

I've tweaked this in https://github.com/AcademySoftwareFoundation/MaterialX/pull/1567/commits/9d4b4929c8bcf5f647692f078e44e7956949ab18, to build the Python bindings when Python API documentation is requested to be built:
```cmake
option(MATERIALX_BUILD_PYTHON_DOCS "Create HTML documentation using Sphinx. Requires that Sphinx be installed." OFF)
if (MATERIALX_BUILD_PYTHON_DOCS)
    set(MATERIALX_BUILD_PYTHON ON)
endif()
```
In the same commit, I've also added `DEPENDS` to the custom `MaterialXDocsPython` target:
```cmake
        add_custom_target(
            MaterialXDocsPython
            ${SPHINX_EXECUTABLE}
                -c ${CMAKE_CURRENT_BINARY_DIR}
                ${SPHINX_SOURCE_DIR}
                ${SPHINX_HTML_OUTPUT_DIR}
            WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
            DEPENDS ${MATERIALX_PYTHON_DOCS_TARGET_DEPENDENCIES}
            COMMENT "Building MaterialX Python API Documentation: ${SPHINX_HTML_OUTPUT_DIR}/index.html"
        )
```
The variable that's referenced in `DEPENDS` is a list of targets for all Python C extension modules:
```cmake
        set(
            MATERIALX_PYTHON_DOCS_TARGET_DEPENDENCIES
            MaterialXCore
            PyMaterialXCore
            PyMaterialXFormat
            PyMaterialXGenShader
            PyMaterialXGenGlsl
            PyMaterialXGenOsl
            PyMaterialXGenMdl
            PyMaterialXGenMsl
            PyMaterialXRender
            PyMaterialXRenderGlsl
            PyMaterialXRenderOsl
            PyMaterialXRenderMsl
        )
```

It seems to work well: when making changes to the docstrings in one of the Python C extensions, only that module is rebuilt, and the docs are regenerated.

I'm using the following command line while iterating on the docs, using `rm` to make sure that the docs are rebuilt from scratch (replaced with `RM` below for safety):
```bash
RM -rf  documents/DeveloperGuide/generated/  documents/html-sphinx/  && cmake --build . --target MaterialXDocsPython
```

While it seems to work, I'm happy to explore alternatives, if this isn't quite hitting the mark.

Reply:
This looks good to me. Thanks for looking into it.